agent-dispatchers/logger/src/main/java/com/expedia/www/haystack/agent/dispatcher/LoggerDispatcher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright 2017 Expedia, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.expedia.www.haystack.agent.dispatcher;
+
+import com.expedia.open.tracing.Span;
+import com.expedia.www.haystack.agent.core.Dispatcher;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.typesafe.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Dispatches the data into stdout.
+ */
+public class LoggerDispatcher implements Dispatcher {
+    private final static Logger LOGGER = LoggerFactory.getLogger(LoggerDispatcher.class);
+
+    @Override
+    public String getName() {
+        return "logger";
+    }
+
+    @Override
+    public void dispatch(final byte[] partitionKey, final byte[] data) {
+        try {
+            Span span = Span.parseFrom(data);
+            LOGGER.debug(span.toString());
```

Comment:
Always calling `toString()` will murder performance.

```java
if (LOGGER.isEnabled(Level.DEBUG)) {
  LOGGER.debug(span.toString());
}
```

It's too bad slf4j is such an old API that it doesn't support something like `LOGGER.debug(span::toString)`

Reply:
Great idea, tho this `LoggerDispatcher` is not meant to be used in prod. Still I think this is the right thing.

Reply:
Yeah now I remember why I did not do 
```
if (LOGGER.isEnabled(Level.DEBUG)) {
  LOGGER.debug(span.toString());
}
```
Which I originally saw in Zipkin code, it was because logback does not support that. In any case I don't think having a test logback config is worth now because matching the exact value of the payload can be problematic because of linebreaks.

Reply:
Oops sorry for slf4j it's isDebugLevelEnabled

But realized this is more idiomatic

`LOGGER.debug("{}", span);`

That makes sure `toString()` is only called when it's enabled. But not sure how this code is used no worries if it's overkill for the use case